File: src/main/java/ru/highloadcup/check/UserEOCustomChecker.java
package ru.highloadcup.check;

import org.springframework.stereotype.Component;
import ru.highloadcup.api.UserEO;

@Component
public class UserEOCustomChecker implements CustomChecker<UserEO> {
    @Override
    public void checkFullyNull(UserEO object) {
        if (object.getEmail() == null
                && object.getBirthDate() == null
                && object.getFirstName() == null
                && object.getLastName() == null
                && object.getGender() == null) {
            throw new IllegalArgumentException("object is fully null");
        }
    }
}

File: src/main/java/ru/highloadcup/check/UserEOCustomChecker.kt
package ru.highloadcup.check

import org.springframework.stereotype.Component
import ru.highloadcup.AllOpen
import ru.highloadcup.api.UserEO

@AllOpen
@Component
class UserEOCustomChecker : CustomChecker<UserEO> {
    override fun checkFullyNull(checkingObject: UserEO) {
        if (checkingObject.email == null
                && checkingObject.birthDate == null
                && checkingObject.firstName == null
                && checkingObject.lastName == null
                && checkingObject.gender == null) {
            throw IllegalArgumentException("checkingObject is fully null")
        }
    }
}
